controller.go

```diff
@@ -50,8 +51,19 @@ func (c *OFController) ConnectionDown() {
 	// handle connection down
 }
 
-func (c *OFController) sendEchoLoop() {
-	// send echo request forever
+func (c *OFController) sendEchoLoop(dp *Datapath) {
+	// send echo request forever, first tick goes after
+	// echo interval
+	fmt.Println("Controller echo loop started for dp: %v", dp)
+	ticker := time.NewTicker(time.Duration(c.echoInterval) * time.Second)
+	defer ticker.Stop()
+	for range ticker.C {
+		echoReq := ofp13.NewOfpEchoRequest()
+		// ToDo: handle graceful shutdown for stale dps as this
+		// go thread will never exit and may be hung on sendMessage
+		// channel.
+		dp.Send(echoReq)
```

Comment:
Given this dp.Send can also block, can we add timed execution for this dp.Send. If this write fails/remains blocked for more than 15 secs, add a panic or at minimum add a log indicating a hang/failure..

Reply:
I would say we should just close the channel for which dp.conn has been closed, and there is already a log before that. That will lead to panic in any thread trying to send to a closed channel or blocked on it

Reply:
you mean add the handling in sendLoop to close channel?

Reply:
Will be doing improvements as follow up PRs.